spring-batch-core/src/main/java/org/springframework/batch/core/repository/support/SimpleJobRepository.java

```diff
@@ -229,7 +229,9 @@ public StepExecution getLastStepExecution(JobInstance jobInstance, String stepNa
 			if (latest == null) {
 				latest = stepExecution;
 			}
-			if (latest.getStartTime().getTime() < stepExecution.getStartTime().getTime()) {
+			// Ordered by step execution ID so less than or equal makes last step execution ID 
+			// the tie breaker if start time is identical
+			if (latest.getStartTime().getTime() <= stepExecution.getStartTime().getTime()) {
```

Comment:
This doesn't actually make the guarantee we need.  The `StepExecutionDao` interface doesn't guarantee order and could be re-implemented for another reason, leaving off the `order by`.  This really should check the `id` to be safe.